compiler/next/include/chpl/uast/Call.h

```diff
@@ -20,40 +20,70 @@
 #ifndef CHPL_UAST_CALL_H
 #define CHPL_UAST_CALL_H
 
-#include "chpl/uast/Exp.h"
-
-#include <vector>
+#include "chpl/uast/Expression.h"
 
 namespace chpl {
 namespace uast {
 
 
 /**
-  This abstract class represents something call-like
+  This abstract class represents something call-like.
+  It represents a called expression as well as a number of actuals.
+
+  For example `f(1,2)`, `f` is the called expression and `1`, `2`
+  are the actuals.
  */
-class Call : public Exp {
+class Call : public Expression {
+ protected:
+  int8_t hasCalledExpression_; // 0 or 1
```

Comment:
That's interesting because I just asked @dlongnecke-cray to change a `bool` into an `int_8`. @e-kayrakli would you feel better if the field is `bool` but when we do index math with it we write something like `int useIdx = idx + (int) hasCalledExpression_;` ?

Reply:
Also this is not a public field, so I don't think we should be *too* concerned about what its declaration type is.

Reply:
Actually, apparently `bool` can implicitly convert to `int` so maybe we can just write `int useIdx = idx + hasCalledExpression_`. I think I was thinking of Chapel's more restrictive implicit conversions.

Reply:
I've changed it to `bool` but left the index math alone.

Reply:
Thanks for changing it, and I am fine with that. It is definitely much better for me to have the field a bool type, and do whatever for the offset expression.

If I were writing this I would do:

```c++
bool hasCalledExpression_;
...

const int offset = hasCalledExpression_ ? 1 : 0;
int useIdx = idx + offset;
```

I am not a big fan of using ints as bools and vice versa.